packages/transaction-controller/src/utils/batch.ts

```diff
@@ -512,3 +540,35 @@ async function processTransactionWithHook(
     params: newParams,
   };
 }
+
+/**
+ * Requests approval for a transaction batch by interacting with the ApprovalController.
+ *
+ * @param txBatchMeta - Metadata for the transaction batch, including its ID and origin.
+ * @param messenger - The messenger instance used to communicate with the ApprovalController.
+ * @param options - Options for the approval request.
+ * @param options.shouldShowRequest - A boolean indicating whether the approval request should be displayed to the user.
+ * @returns A promise that resolves to the result of adding the approval request.
+ */
+async function requestApproval(
+  txBatchMeta: TransactionBatchMeta,
+  messenger: TransactionControllerMessenger,
+  { shouldShowRequest }: { shouldShowRequest: boolean },
```

Comment:
As we only have one usage, should we remove this property and hardcode below?

Reply:
Done